Frameworks/AddressBook/ABMultiValueInternal.mm

```diff
@@ -159,4 +159,71 @@ - (CFIndex)getIndexForIdentifier:(ABMultiValueIdentifier)identifier {
     return index;
 }
 
+- (_ABMultiValue*)createMutableCopy {
+    _ABMultiValue* copy = [[_ABMultiValue alloc] initWithPropertyType:[self getPropertyType]];
+    for (__ABMultiValuePair* pair in self->_list) {
+        [copy appendPairWithLabel:pair.label andValue:pair.value];
+    }
+
+    return copy;
+}
+
+- (bool)addValue:(CFTypeRef)value andLabel:(CFStringRef)label outIdentifier:(ABMultiValueIdentifier*)outIdentifier {
+    bool result = [self appendPairWithLabel:(__bridge NSString*)label andValue:(__bridge id)value];
+    if (result && outIdentifier) {
+        *outIdentifier = [self getIdentifierAtIndex:([self getCount] - 1)];
```

Comment:
The multi-value documentation itself doesn't talk about thread safety, but Apple does say this about working with an AddressBook in general: "You must ensure that an instance of ABAddressBookRef is used by only one thread."

---
In reply to: [75726420](https://github.com/Microsoft/WinObjC/pull/893#discussion_r75726420) [](ancestors = 75726420,75710964,75513394)